utils/rate.go

```diff
@@ -55,45 +52,37 @@ func (rlt *rateLimitTransport) RoundTrip(req *http.Request) (*http.Response, err
 		return resp, err
 	}
 
-	// Make response body accessible for retries & debugging
-	// (work around bug in GitHub SDK)
-	// See https://github.com/google/go-github/pull/986
-	r1, r2, err := drainBody(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-	resp.Body = r1
-	ghErr := github.CheckResponse(resp)
-	resp.Body = r2
-
-	// When you have been limited, use the Retry-After response header to slow down.
-	if arlErr, ok := ghErr.(*github.AbuseRateLimitError); ok {
```

Comment:
do you think it makes sense to handle abuse as well? We don't need it right now because of the current logic in shallow command but we might hit it later. The check is simple: https://github.com/google/go-github/blob/master/github/github.go#L754

Reply:
yup, I didn't add it because it wouldn't be useful the `shallow` command, but given its simplicity I guess that we can include it in this PR.

Reply:
Reincluded abuse check [here](https://github.com/src-d/ghsync/pull/51/commits/5acc8d7146da5cd962bf5f1c53ba48aaae011790).